### ISSUE TITLE
"tourism" refrains "man_made" node from rendering "name"

### ISSUE THREAD

**Issue by IgorEliezer** [gravitystorm/openstreetmap-carto#3545]
### Expected behavior

Related to: https://www.openstreetmap.org/node/1980961590

Combining `man_made=cross` and `tourism=attraction` should not refrain the feature from rendering its `name` beneath the cross icon.

### Actual behavior

See history: https://www.openstreetmap.org/node/1980961590/history

At the version 1, the node had only `tourism=attraction` and `name=Morro da Santa`. I remember its name rendered at the node.

Once I added `man_made=cross`, it started to render the cross icon but, on the other hand, its name no longer renders.

Now, compare with https://www.openstreetmap.org/node/5185486387 which has `man_made=cross` and `name`. Its name is rendered beneath the cross icon. My guess is that, the problem is on `tourism=attraction` side.

### Links and screenshots illustrating the problem

![map](https://user-images.githubusercontent.com/4673289/49333043-ff337380-f59e-11e8-95ec-3c9fe59e5181.png)

**Comment by kocio-pl** [comment]
Thanks for the report, it looks like a valid problem. Would you like to prepare the code fixing this?

**Comment by jeisenbe** [comment]
This is a text-dy problem.

If there is a man_made=cross alone, the code moves the name label text down by 6 pixels so that the name label is not blocked by the icon:

```
    [feature = 'man_made_cross'],
    [feature = 'historic_wayside_cross'] {
      text-dy: 6;
```

But tourism=attraction is rendered without an icon, like an area feature, the same as zoos and theme parks:

```
      [feature = 'tourism_attraction'] {
        text-fill: #660033;
        text-face-name: @standard-font;
      }
```

The wiki page for tourism=attraction says:

"This tag may be added to object to indicate that the place is interesting for tourists. Note that tagging just tourism=attraction is not enough - this tag should be used only as addition to main tag describing the feature.

"For example tourism=attraction + name=Waterfall "Foobar" is not a proper way to tag waterfall named Foobar, such object should be tagged rather with waterway=waterfall + tourism=attraction + name=Foobar."

"Other (rarely used) possible way is to map the primary feature (eg waterfall) and add tourism=yes."

So this suggests that we could solve this problem by removing the code to render names for features that are tagged tourism=attraction without another tag.

Thoughts on this option?

**Comment by jeisenbe** [comment]
Previous threads and comments about tourism=attraction :

https://github.com/gravitystorm/openstreetmap-carto/issues/460#issuecomment-58790797
https://github.com/gravitystorm/openstreetmap-carto/issues/1029#issuecomment-68574622

#1047 "Change color tourism=attraction label and drop rendering of tourism=attraction areas "
#1063 "tourism=attraction - partially restore old label, stop rendering areas"
#1257 "Add area render for tourism=attraction"

**Comment by Adamant36** [comment]
That does seem to the issue as other things co-tagged with tourism attraction dont seem to have their names render either. Whereas places just tagged as tourism attraction do have their names rendered. The problem with removing name rendering from attraction is that you'll be removing a lot of names of importent features from the map. As a lot of things are tagged that way despite the wiki. Id imagine there's a lot of places out there like city districts or wharfs that dont have clear boundaries or objects to add the tourism node to that are still interesting to tourists. There's a few such places in San Francisco. It would be a shame to remove the names of those things from the map. Although, im not sure there is a better solution, but its still something to consider.

**Comment by dieterdreist** [comment]
sent from a phone

> On 4. Dec 2018, at 02:53, jeisenbe <notifications@github.com> wrote:
> 
> So this suggests that we could solve this problem by removing the code to render names for features that are tagged tourism=attraction without another tag.
> 
> Thoughts on this option?
> 


I am in favor. Many people look at osm-carto to check their mapping, if the attraction tag alone doesn’t lead to renderer they will often add more (descriptive) tags.

**Comment by HolgerJeromin** [comment]
ref #3446

**Comment by jeisenbe** [comment]
This tag has now been discussed on the tagging mailing list. https://lists.openstreetmap.org/pipermail/tagging/2018-December/

Many people felt that all tourism=attraction can be tagged with a more specific tag. Taginfo shows a number of popular combinations https://taginfo.openstreetmap.org/tags/tourism=attraction#combinations

However, there are over 10,000 combined with attraction=*, mostly attraction=animal (to tag a specific animal exhibit in a zoo). Attraction is also frequently used with water_slide, roller_coaster, amusement_ride, and carousel. These seem to be fairly reasonable uses of the tourism=attraction tag, though perhaps attraction=* alone would be sufficient. 
https://taginfo.openstreetmap.org/keys/attraction

While we could switch to rendering only features also tagged with attraction=*, it would be less disruptive to switch to rendering tourism=attraction only at z17 and higher, and no longer rendering it early when tagged on large areas. A generic marker should also be added to show the location of the attraction. I'd suggest using @tourism color to match zoos and amusement parks.

This will appropriately render attractions such as animals in zoos and amusement rides in theme parks, but should encourage mappers to look for better ways to tag large areas.

I found that I can resolve the initial issue by moving 'tourism' below man_made_cross in project.mml, so that it renders after man_made=cross and historic=wayside_cross. 

**Test area, currently:**
("bench attraction" is a man_made=bench tagged with tourism=attraction. "Animal attraction" is tourism=attraction with attraction=animal. "Baliem Resort" is a leisure=resort also tagged tourism=attraction. The two crosses without names shown are also tagged tourism=attraction)
z16
![z16-tourism-attraction-before](https://user-images.githubusercontent.com/42757252/50225324-8b3cfc00-03e3-11e9-9d42-97e9f82cf1c3.png)

z15
![z15-tourism-attraction-before png](https://user-images.githubusercontent.com/42757252/50225338-9728be00-03e3-11e9-9ba1-d4b73d7c49f7.png)

**After**
z16
![z16-tourism-attraction-after](https://user-images.githubusercontent.com/42757252/50225352-9c860880-03e3-11e9-9832-f74ee54bb575.png)

z15
![z15-tourism-attraction-after](https://user-images.githubusercontent.com/42757252/50225362-a60f7080-03e3-11e9-8f90-19ed6ca6f1d9.png)

**Comment by Adamant36** [comment]
@jeisenbe, how come you do test renderings with style maps (or whatever their called) instead of using the actual map map in Kosmtik? I noticed you did the same thing with roads. It would be interesting to know if it might give different results that way or what the pros/cons are. I'm sure whatever your using is a lot smaller in file size then the average shape file at least.

**Comment by kocio-pl** [comment]
I still think violet is good for both attraction and zoo/theme park.

**Comment by jeisenbe** [comment]
> how come you do test renderings with [test data] instead of using the
actual map data?

As you guessed, it saves internet data. In this case I had already added
these features in JOSM while trying to find if this issue occurred with
other features, or just with man_made=cross and historic=wayside_cross, so
it was easy to show a test image this way

For highway=construction it is impossible to find all types of
construction=* in one area. It is much easier to test with artificial data.

> I still think violet is good for both attraction and zoo/theme park.

I agree!

Should we do that PR first or do we need to change the boundary color first?

**Comment by kocio-pl** [comment]
I would start with boundary. That would be also needed for merging #3521 at last.

**Comment by matthijsmelissen** [comment]
I still don't fully understand the problem in this issue - could somebody give me a short summary of the cause?

**Comment by jeisenbe** [comment]
I believe I understand the problem.

Man_made=cross and historic=wayside cross have name renderings, but they
are in the low-priority amenities layer

Tourism=attraction is in the regular layer, which is rendered first (or at
higher priority?) based on its position in project.mml

But tourism=attraction features currently render as maroon text without an
icon. There is no text-dy so the cross icon blocks the name, which would
otherwise render at the same place. (I’ve seen this clearly at meta tile
boundaries while testing)

So to fix this, I’ve recommended we
move tourism_attraction to the low-priority layer
- this is sensible because it is often a secondary tag.
- but it is also sometimes a feature, eg a rollercoaster or a zoo animal,
and should render when there is no other tag.

**Comment by matthijsmelissen** [comment]
@jeisenbe I still don’t follow. Which feature blocks rendering of the man_made cross name?

**Comment by matthijsmelissen** [comment]
Ah, I guess I get it now. So part of the issue is that for objects that are both man_made=cross and tourist=attraction, the label is rendered in the latter style? Perhaps we could solve this by change this order?

**Comment by jeisenbe** [comment]
Right. It was only a problem with man_made=cross and
historic=wayside_cross, because these are rendered at lower priority and do
have a name label.

In the PR I moved the tourism_attraction line down below the other features
in the text-point and text-poly layers in project.mml and this fixed the
issue.

Actually, I just remembered that it isn’t necessary to have tourism
attraction in the amenity_low_priority layers now, because we are not going
to render the icon. So I’ve removed these two lines from project.mml
